kernel/udsdev.c

```diff
@@ -1324,6 +1340,14 @@ static int _udsdev_setsockopt(
             _obj(sock)->so_rcvbuf = _max(_obj(sock)->so_rcvbuf, MIN_SO_RCVBUF);
             break;
         }
+        case SO_SNDTIMEO:
+        {
+            if (!optval)
+                ERAISE(-EINVAL);
+
+            memcpy(&_obj(sock)->so_sndtimeo, optval, sizeof(struct timeval));
```

Comment:
Add check that ``optlen == sizeof(struct timeval)`` and fail if not. This can result in a free memory read if the caller passes in something smaller than ``struct timeval``.

Reply:
Also should fail if optlen is null.

Reply:
Added a check to fail if `optlen < sizeof(struct timeval)`.